sentry_sdk/integrations/django/signals_handlers.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from django.dispatch.dispatcher import NO_RECEIVERS
+from sentry_sdk import Hub
+
+from django.dispatch import Signal
+
+
+def patch_signals():
+    """Patch django signal receivers to create a span"""
+
+    def send(self: Signal, sender, **kwargs):
```

Comment:
Here you basically replace the existing `send()` function with your own version. 

Best practice is, that we just wrap the existing function and run our code before or after the existing function. 

See this as a template: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/starlette.py#L204-L218

Reply:
yeah I wanted to originally just wrap the existing function for improved compatibility, but with that there's no way to instrument individual signal handlers

Reply:
I guess it can work by patching `_live_receivers()` and instead of returning an array of receivers (=functions) we wrap all the elements of the array in a function that creates a span and then calls the old receiver.

Something like this:
```python
old_live_receivers = Signal._live_receivers

def _sentry_live_receivers(self, sender)
    receivers = old_live_receivers(self, sender)

    def _receiver_with_span(*args, **kwargs):
        old_receiver = receivers[idx]
        name = old_receiver.__name__
        with hub.start_span(    
            op="django.signals",
            description=name,
        ) as span:
            span.set_data("signal", name)
            return old_receiver(*args, **kwargs)
            
    for idx, receiver in enumerate(receivers):
        receivers[idx] = _receiver_with_span

    return receivers

Signal._live_receivers = _sentry_live_receivers
```

Note: this is untested code, I just wrote this on top of my head without ever running it. But I did something similar in another integration a couple of days ago, so it might work.

Reply:
yeah thats a good idea actually